build/source/engine/vegNrgFlux.f90

```diff
@@ -581,70 +519,58 @@ subroutine vegNrgFlux(&
  scalarLatHeatGround             => flux_data%var(iLookFLUX%scalarLatHeatGround)%dat(1),            & ! intent(out):   [dp] latent heat flux from ground surface below vegetation (W m-2)
-
  ! output: advective heat fluxes
  scalarCanopyAdvectiveHeatFlux   => flux_data%var(iLookFLUX%scalarCanopyAdvectiveHeatFlux)%dat(1),  & ! intent(out): [dp] heat advected to the canopy surface with rain + snow (W m-2)
  scalarGroundAdvectiveHeatFlux   => flux_data%var(iLookFLUX%scalarGroundAdvectiveHeatFlux)%dat(1),  & ! intent(out): [dp] heat advected to the ground surface with throughfall (W m-2)
-
  ! output: mass fluxes
  scalarCanopySublimation         => flux_data%var(iLookFLUX%scalarCanopySublimation)%dat(1),        & ! intent(out): [dp] canopy sublimation/frost (kg m-2 s-1)
  scalarSnowSublimation           => flux_data%var(iLookFLUX%scalarSnowSublimation)%dat(1),          & ! intent(out): [dp] snow sublimation/frost -- below canopy or non-vegetated (kg m-2 s-1)
-
  ! input/output: canopy air space variables
  scalarVP_CanopyAir              => diag_data%var(iLookDIAG%scalarVP_CanopyAir)%dat(1),             & ! intent(inout): [dp] vapor pressure of the canopy air space (Pa)
  scalarCanopyStabilityCorrection => diag_data%var(iLookDIAG%scalarCanopyStabilityCorrection)%dat(1),& ! intent(inout): [dp] stability correction for the canopy (-)
  scalarGroundStabilityCorrection => diag_data%var(iLookDIAG%scalarGroundStabilityCorrection)%dat(1),& ! intent(inout): [dp] stability correction for the ground surface (-)
-
  ! output: liquid water fluxes
  scalarCanopyTranspiration       => flux_data%var(iLookFLUX%scalarCanopyTranspiration)%dat(1),      & ! intent(out): [dp] canopy transpiration (kg m-2 s-1)
  scalarCanopyEvaporation         => flux_data%var(iLookFLUX%scalarCanopyEvaporation)%dat(1),        & ! intent(out): [dp] canopy evaporation/condensation (kg m-2 s-1)
  scalarGroundEvaporation         => flux_data%var(iLookFLUX%scalarGroundEvaporation)%dat(1),        & ! intent(out): [dp] ground evaporation/condensation -- below canopy or non-vegetated (kg m-2 s-1)
-
  ! output: derived fluxes
  scalarTotalET                   => flux_data%var(iLookFLUX%scalarTotalET)%dat(1),                  & ! intent(out): [dp] total ET (kg m-2 s-1)
  scalarNetRadiation              => flux_data%var(iLookFLUX%scalarNetRadiation)%dat(1)              & ! intent(out): [dp] net radiation (W m-2)
- )
+ ) ! end associate statement
  ! ---------------------------------------------------------------------------------------
- ! initialize error control
- err=0; message="vegNrgFlux/"
+ err=0; message="vegNrgFlux/" ! initialize error control
 
- ! initialize printflag
- printflag = .false.
-
- ! identify the type of boundary condition for thermodynamics
- select case(ix_bcUpprTdyn)
+ printflag = .false. ! initialize printflag
```

Comment:
If (big if) we remove all the debug printing statements that use `printFlag`, should we delete it here too or does it serve multiple purposes?

Reply:
If it is deemed acceptable to permit removal of the debug statements, then we can remove the declaration and initialization of the `printflag` logical. I have removed it for now, to be consistent with the rest of the develop_refactor branch. However, if it is decided that the debug statements should be retained, I will restore the logical.